### PR TITLE
Explicit queue flush

### PR DESCRIPTION
**Pull request by priyaashah**
Flush analytics items queue explicitly when:
1. Button is attached to window
2. Button is detached from window
4. Queue threshold is reached.

This is to make sure we don't lose any tracked events

### CODE REVIEW

File: connect-button/src/main/java/com/ifttt/connect/ui/BaseConnectButton.java [hunk 1]
```diff
@@ -183,13 +181,16 @@ public BaseConnectButton(Context context, @Nullable AttributeSet attrs, int defS
     protected void onAttachedToWindow() {
         super.onAttachedToWindow();
         lifecycleRegistry.markState(STARTED);
+        AnalyticsManager.getInstance(getContext().getApplicationContext())
```
Comment (lzanita09) [inline connect-button/src/main/java/com/ifttt/connect/ui/BaseConnectButton.java]: `getInstance` is already calling `getApplicationContext`.

File: connect-button/src/main/java/com/ifttt/connect/ui/BaseConnectButton.java [hunk 2]
```diff
@@ -1137,6 +1143,17 @@ public void onActivityResumed(Activity activity) {
                     activityLifecycleCallbacks = null;
                 }
             }
+
+            @Override
+            public void onActivityPaused(Activity activity) {
+                /*
+                * Clear the event queue when activity is paused to make sure the queued events are not missed
+                * */
+                if (activity == context) {
```
Comment (lzanita09) [inline connect-button/src/main/java/com/ifttt/connect/ui/BaseConnectButton.java]: I don't think we need to be this aggressive sending the batched events, especially this callback is only used when the webview is open.

Comment (priyaashah) [reply connect-button/src/main/java/com/ifttt/connect/ui/BaseConnectButton.java]: We may lose the connect button click event for certain scenarios if we don't flush events here.

Comment (priyaashah) [reply connect-button/src/main/java/com/ifttt/connect/ui/BaseConnectButton.java]: Talked irl, and we don't need this aggressive queue flush here.

**Pull request merged**